zabbix/resource_zabbix_item.go

```diff
@@ -0,0 +1,263 @@
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"item_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "(readonly) ID of the item.",
+			},
+			"delay": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"host_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the host or template that the item belongs to.",
+			},
+			"interface_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "0",
+			},
+			"key": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Item key.",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the item.",
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 0 || v > 16 {
+						errs = append(errs, fmt.Errorf("%q, must be between 0 and 16 inclusive, got %d", key, v))
+					}
+					return
+				},
+			},
+			"value_type": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 0 || v > 4 {
+						errs = append(errs, fmt.Errorf("%q, must be between 0 and 4 inclusive, got %d", key, v))
+					}
+					return
+				},
+			},
+			"data_type": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 0 || v > 3 {
+						errs = append(errs, fmt.Errorf("%q, must be between 0 and 3 inclusive, got %d", key, v))
+					}
+					return
+				},
+			},
+			"delta": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Value that will be stored. ",
+				Default:     0,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 0 || v > 2 {
+						errs = append(errs, fmt.Errorf("%q, must be between 0 and 2 inclusive, got %d", key, v))
+					}
+					return
+				},
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the item. ",
+				Default:     "",
+			},
+			"history": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Number of days to keep item's history data. Default: 90. ",
+				Default:     "90",
+			},
+			"trends": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Number of days to keep item's trends data. Default: 365. ",
```

Comment:
Same as for `history`.

Reply:
Description needs to be updated accordingly to latest change.

Reply:
```suggestion
				Description: "Number of days to keep item's trends data. Default: `365d` with Zabbix Server version 3.4.0+, `365` otherwise.",
```